### PR TITLE
Fix: Request money - Five most recent chats aren't shown correctly

### PR DESCRIPTION
**Pull request by metehanozyurt**
### Details
getNewChatOptions and getSearchOptions functions changed for passing sortByLastMessageTimestamp parameter. NewChatPage.js , OUParticipantsRequest.js , IOUParticipantsSplit.js files changed. "New chat" , "New group" , "Send money" , "Request money" , "Split bill" pages recents chat areas fixed.

### Fixed Issues

$ https://github.com/Expensify/App/issues/8220

### Tests

1. Open Expensify App.
2. Click left side chat area randomly
3. Click plus button and click "Request money". Fill number area then click "next"
4. Request money page `Recents` section should be ordered last message time ,`Contacts` section should be sorted by alphabetical order
5. Click plus button and click "New chat".
6. New chat page `Recents` section should be ordered last message time ,`Contacts` section should be sorted by alphabetical order
7. Click plus button and click "Send money". Fill number area then click "next"
8. Send money page `Recents` section should be ordered last message time ,`Contacts` section should be sorted by alphabetical order
9. Click plus button and click "Split bill". Fill number area then click "next"
10. Split bill page `Recents` section should be ordered last message time ,`Contacts` section should be sorted by alphabetical order
11. Click plus button and click "New group".
12. New group page `Recents` section should be ordered last message time ,`Contacts` section should be sorted by alphabetical order
13. Click profile button, Click workspaces, click manage members, click invite button. verify persons not sorted by alphabetical order

example reproduce issue video before fix:

https://user-images.githubusercontent.com/6093207/162324211-ce370681-e5a6-4d85-b03b-fe6c3f5b97dd.mp4

test video:

https://user-images.githubusercontent.com/6093207/162325410-4cdb3a8e-955c-41f3-b1f3-62149e04294e.mp4

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [ ] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps

1. Open Expensify App.
2. Click left side chat area randomly
3. Click plus button and click "Request money". Fill number area then click "next"
4. Request money page `Recents` section should be ordered last message time ,`Contacts` section should be sorted by alphabetical order
5. Click plus button and click "New chat".
6. New chat page `Recents` section should be ordered last message time ,`Contacts` section should be sorted by alphabetical order
7. Click plus button and click "Send money". Fill number area then click "next"
8. Send money page `Recents` section should be ordered last message time ,`Contacts` section should be sorted by alphabetical order
9. Click plus button and click "Split bill". Fill number area then click "next"
10. Split bill page `Recents` section should be ordered last message time ,`Contacts` section should be sorted by alphabetical order
11. Click plus button and click "New group".
12. New group page `Recents` section should be ordered last message time ,`Contacts` section should be sorted by alphabetical order
13. Click profile button, Click workspaces, click manage members, click invite button. verify persons not sorted by alphabetical order

- [ ] Pages that changes recents order (sort by last message time):
    - [ ] New chat page (`Recents` section)
    - [ ] New group page (`Recents` section)
    - [ ] Send money page (`Recents` section)
    - [ ] Request money page (`Recents` section)
    - [ ] Split bill page (`Recents` section)
- [ ] Pages that changes contacts order (sort by alphabetically order):
    - [ ] New chat page (`Contacts` section)
    - [ ] New group page (`Contacts` section)
    - [ ] Send money page (`Contacts` section)
    - [ ] Request money page (`Contacts` section)
    - [ ] Split bill page (`Contacts` section)

- [ ] Verify that no errors appear in the JS console

#### Web

https://user-images.githubusercontent.com/6093207/173229367-1c658c93-0987-467a-b37b-91db3a87a7c0.mp4

#### Mobile Web

https://user-images.githubusercontent.com/6093207/173229396-9c97ef13-8c51-4a40-b8ed-5dbfac111008.mp4

https://user-images.githubusercontent.com/6093207/173229402-d634aada-969c-45c7-83d7-b5d2c6dc5bb4.mp4

#### Desktop

https://user-images.githubusercontent.com/6093207/173229421-4e413e70-94f2-4585-abb0-e8a677f22624.mp4

#### iOS

https://user-images.githubusercontent.com/6093207/173229442-62f759e3-5b9f-40e8-a717-b9e756473a68.mp4

#### Android

https://user-images.githubusercontent.com/6093207/173229451-19a72d40-1fed-424c-b798-d4da2db4b084.mp4

### CODE REVIEW

**Comment by marcaaron** [comment]
Just leaving a quick comment, but I think it would be easier to default `sortByLastMessageTimestamp` to `true` for `getNewChatOptions()` and then specifically pass `false` for the "workspace members" usages if we want those not to be sorted.

**Review by parasharrajat — CHANGES_REQUESTED** [review]
We don't have to pass the config options from the callable location. The whole purpose of creating `getNewChatOptions`, `getSearchOptions` functions was to encapsulate the configuration and use it as it is. 

Secondly, it seems every function is setting that option to `true`. SO set it to `true` by default value and then should be enough. Then explicitly set it false where needed.

**Comment by metehanozyurt** [comment]
Thank you so much @parasharrajat , @marcaaron . I will immediately update commit and testDocument for changes 🙏.

**Comment by parasharrajat** [comment]
Need to confirm the sort order for various pages. Thanks for the patience.

**Comment by parasharrajat** [comment]
@metehanozyurt Could you please merge the main so that I can test it?

**Comment by metehanozyurt** [comment]
Now I merged the conflict problem solved. So Sorry for falling behind the main.

**Comment by parasharrajat** [comment]
I will review this in some time.

File: src/pages/SearchPage.js [hunk 1]
```diff
@@ -65,6 +65,7 @@ class SearchPage extends Component {
             props.personalDetails,
             '',
             props.betas,
+            true,
```
Comment (parasharrajat) [inline src/pages/SearchPage.js:68]: we should not expose the config option outside. Instead of doing this create another wrapper method like `OptionsListUtils#getSearchOptions` which sets this flag to true and then use that method.

File: tests/unit/OptionsListUtilsTest.js [hunk 2]
```diff
@@ -295,12 +295,44 @@ describe('OptionsListUtils', () => {
         expect(results.recentReports[0].text).toBe('The Flash');
     });
 
-    it('getNewChatOptions()', () => {
+    it('getSearchOptions() with message timeStampOrder', () => {
```
Comment (parasharrajat) [inline tests/unit/OptionsListUtilsTest.js:298]: When you will do the above change, here new test will be created for each function including that new function.

**Review by parasharrajat — CHANGES_REQUESTED** [review]
let me know when ready so that I can test.

File: src/pages/SearchPage.js (hunk 1 above)
Comment (metehanozyurt) [reply src/pages/SearchPage.js:68]: After new updates all places using functions started sending true by default (getSearchOptions & getNewChatOptions). Now we dont need wrapper method. Because If I write a new function, the old one will not be used this time. I'm making the changes now as you said. I'm removing the parameter pass support. Thank you. 
After this i will update PR videos immediately.

File: tests/unit/OptionsListUtilsTest.js (hunk 2 above)
Comment (metehanozyurt) [reply tests/unit/OptionsListUtilsTest.js:298]: I will update OptionsListUtilsTest test and removing condition timeStampOrder (true-false). Thank you.

**Comment by metehanozyurt** [comment]
https://github.com/Expensify/App/pull/8549#discussion_r862389693
I did not write a new wrapper method for the reason I mentioned here. I reverted the parameters it received.
I updated PR videos and checked results. I added lines to check the alphabet to the steps in the test file. I hope I succeeded this time.

There is one last thing I want to say @parasharrajat . When I examined the PR files, I realized that it is more beautiful now. Thank you very much for your guidance and help. Sorry for not being able to do it earlier. I hope I didn't seem too emotional, but I really thank you 🙏 .

File: src/pages/SearchPage.js (hunk 1 above)
Comment (parasharrajat) [reply src/pages/SearchPage.js]: Sorry, But I didn't understand. Could you please rephrase this?

Comment (metehanozyurt) [reply src/pages/SearchPage.js]: `getSearchOptions` function only used on SearchPage. That's why I fixed the mistake I made by getting extra config. if i wrote another function the old function would be un used.

**Comment by parasharrajat** [comment]
I will review this today. Apologies for the delay. It just got out of sync.

File: tests/unit/OptionsListUtilsTest.js [hunk 3]
```diff
@@ -283,9 +283,10 @@ describe('OptionsListUtils', () => {
         // When we filter again but provide a searchValue that should match multiple times
         results = OptionsListUtils.getSearchOptions(REPORTS, PERSONAL_DETAILS, 'fantastic');
 
-        // Then we get both values with the pinned value still on top
+        // lastMessageTimestamp value should be top
```
Comment (parasharrajat) [inline tests/unit/OptionsListUtilsTest.js:286]:
```suggestion
        // Value with latestd lastMessageTimestamp should be at the top.
```

File: tests/unit/OptionsListUtilsTest.js [hunk 4]
```diff
@@ -309,13 +310,28 @@ describe('OptionsListUtils', () => {
...
+        expect(results.personalDetails[3].text).toBe('The Incredible Hulk');
+
         // Then the result which has an existing report should also have the reportID attached
         const personalDetailWithExistingReport = _.find(
             results.personalDetails,
             personalDetail => personalDetail.login === 'peterparker@expensify.com',
         );
         expect(personalDetailWithExistingReport.reportID).toBe(2);
 
+        // When we call only pass personal details
```
Comment (parasharrajat) [inline tests/unit/OptionsListUtilsTest.js:326]:
```suggestion
        // When we only pass personal details
```

**Review by parasharrajat — CHANGES_REQUESTED** [review]

**Comment by metehanozyurt** [comment]
I made the changes you mentioned. Thank you for your patience and helps @parasharrajat 🙏 .

**Comment by parasharrajat** [comment]
@metehanozyurt 
We are close to the end, the only change that is left is to correct the WorkspaceInvitePage. it should be ordered in the most recent first. I was pushing this since the start but due to multiple update requests, it slipped my mind.  

I can not remember any other page that is an exception to the default ordering except `WorkspaceInvitePage`. Let me know if find any.

In the QA steps. Please mention testing each page whose order is changed. Make a checklist using the following syntax. 
```
 - [ ] page 
```

**Comment by metehanozyurt** [comment]
I am so sorry about the delay. I was very sick this week , I feel a little better today. I will complete it today. Have a nice day @parasharrajat .

**Comment by parasharrajat** [comment]
No worries. Take rest and get better.

**Comment by metehanozyurt** [comment]
I want to consult about a subject before preparing a new commit @parasharrajat .
When we most recent first on the WorkspaceInvitePage, users seem to be missing. Because he may not have spoken to the user in the contact list yet. 

I have a suggestions about it. 
1-) After adding them ordered in the most recent first, can I sort and add user who have not spoken yet but are in the contact list according to the alphabet? 
2-) I can change to two section `RESENTS` and `CONTACTS` like the other pages(`New chat page`, `New group page`).

I wanted to ask this because there are not two separate sections like `New chat page`. I have concerns that users may missing compared to the previous version. 

Also New chat , New group , Send money , Request money and Split bill contacts lists changed. 
Previously, It was displayed at the added order to Onyx (`ONYXKEYS.PERSONAL_DETAILS`). But now it sorts alphabetically. Can I mention this in the QA steps?

I created a new wrapper method (`OptionsListUtils#getMemberInviteOptions`) for the workspace invite page because I need to add one parameter different (`maxRecentReportsToShow: 0 , // Unlimited`) . For your information, I'm writing a test for this as well.

**Comment by parasharrajat** [comment]
> Also New chat , New group , Send money , Request money and Split bill contacts lists changed.
Previously, It was displayed at the added order to Onyx (ONYXKEYS.PERSONAL_DETAILS). But now it sorts alphabetically. Can I mention this in the QA steps?

yes


> When we most recent first on the WorkspaceInvitePage, users seem to be missing. Because he may not have spoken to the user in the contact list yet.

Why? Wouldn't it be just `sortByLastMessageTimestamp:false` for `OptionsListUtils#getMemberInviteOptions`. Do you know why this is happening?

**Comment by metehanozyurt** [comment]
I think I know. The reason why we encountered this situation is as follows.
We need to `recentReports` for sorted recent contacts.This list consists of people from the `reports` object.
But there may also be people who do not have a report object. These people return in the personalDetails list.

Before The reports object was not given because there was no prior sorting. For this reason, the personalDetails object consisted of all contacts.

Before :
https://github.com/Expensify/App/blob/0609796c7e0d3768217bc381b7320420c41b27d6/src/pages/workspace/WorkspaceInvitePage.js#L68-L78

After I planning:
```js
const {
            personalDetails,
            userToInvite,
            recentReports,
        } = OptionsListUtils.getMemberInviteOptions(
            props.reports,
            props.personalDetails,
            props.betas,
            '',
            [],
            this.getExcludedUsers(),
        );
```


I think we should combine these two lists (`recentReports`,`personalDetails`). 
Can I make a new commit ? I think you might like this idea when you see it 👼 .

**Comment by parasharrajat** [comment]
No, reports are not supposed to be part of this list. Only contacts.

**Comment by metehanozyurt** [comment]
Ok , We want to list of contacts like LHN but only contacts. Recents on the top am i right? We need `reports` to sort contacts because it contains the lastMessageTimestamp information. If we do not give this, we cannot make a sort. 

For Example New Chat Page;
This page give us 5 recent contacts.
`recentReports` represent 5 recent contacts.
https://github.com/Expensify/App/blob/fc3e13ea735874b4223b3ce2a88cc03e83ec9ee7/src/pages/NewChatPage.js#L65-L76
If we don't give the reports and only empty array(line 70), it will only give us the contact list(alphabeticly ordered). The `recentReports` information will never come.

**Comment by parasharrajat** [comment]
Ok, I would say just leave the contacts in A-Z order. 

@roryabraham Do you think the A-Z order is fine for contacts on the Workspace member invite page?

**Comment by roryabraham** [comment]
> Do you think the A-Z order is fine for contacts on the Workspace member invite page?

Yes I think that would be fine in isolation – but for search, new chat, and request pages we'll want to sort results by most recent timestamp. Given that, I would say let's be consistent for the workspace member invite page.

**Comment by parasharrajat** [comment]
Ok so @metehanozyurt we can just keep the workspace behavior the same as it was before. The only change would be that previously it will now be sorted using `sortByLastMessageTimestamp`.

**Comment by metehanozyurt** [comment]
Thank you @parasharrajat . I updated the QA steps and created a checklist. I hope it was in the requested format and descriptive.

**Comment by parasharrajat** [comment]
Thanks for doing that. Do you have an update for https://github.com/Expensify/App/pull/8549#issuecomment-1138335682?

**Comment by parasharrajat** [comment]
I am following this https://github.com/Expensify/App/pull/8549#issuecomment-1137988535. Please feel free to open this on Slack to get consensus.

**Comment by parasharrajat** [comment]
Did you get a chance to ask this on slack?

**Comment by metehanozyurt** [comment]
Thanks @parasharrajat for chance to do this. I asked a question, I hope I did it right. I was a little nervous as this was my first time doing something like this.
https://expensify.slack.com/archives/C01GTK53T8Q/p1654330306672439

**Comment by metehanozyurt** [comment]
I'm a little sorry that my question didn't get much attention 😢.
Would you like me to leave the Workspace invite page as it is currently live version on web?

I can't sort by last message time information because it will only be a list of people messaged before.
A-Z order was also not considered appropriate.
The only option I have left is to write a new wrapper method and show it without A-Z sorting contacts.

**Comment by parasharrajat** [comment]
I have bumped the thread.

**Comment by metehanozyurt** [comment]
Thank you for helping me on Slack @parasharrajat .
For this i need to one parameter for this so i add `sortPersonalDetailsByAlphaAsc` in getOptions params its default true . And write new wrapper method `getMemberInviteOptions`.  
I couldn't use the `sortByAlphaAsc` parameter because it was used to sort reports. Reports are not sent on our page (Workspace Invite Page).
```js
function getMemberInviteOptions(
    reports,
    personalDetails,
    betas = [],
    searchValue = '',
    selectedOptions = [],
    excludeLogins = [],
) {
    return getOptions(reports, personalDetails, 0, {
        betas,
        searchValue: searchValue.trim(),
        selectedOptions,
        excludeDefaultRooms: true,
        includePersonalDetails: true,
        excludeLogins,
        sortPersonalDetailsByAlphaAsc: false,
    });
}
```

I am currently updating the test document for this new method.
I'll update the videos once I'm done with my work. Thank you again about the Slack topic 🙏 .

**Comment by metehanozyurt** [comment]
Hi @parasharrajat , I merged it with main branch. I updated test file , PR steps and videos. Thanks for your time 🙏 .

File: src/libs/OptionsListUtils.js [hunk 5]
```diff
@@ -607,7 +613,6 @@ function getOptions(reports, personalDetails, activeReportID, {
             return 0;
         }], ['asc']);
     }
-
```
Comment (parasharrajat) [inline src/libs/OptionsListUtils.js:610]: NAB: unnecessary change.

File: src/libs/OptionsListUtils.js [hunk 6]
```diff
@@ -711,6 +715,36 @@ function getNewChatOptions(
...
+ * @param {Object} reports
+ * @param {Object} personalDetails
+ * @param {Array<String>} betas
+ * @param {String} searchValue
+ * @param {Array} selectedOptions
+ * @param {Array} excludeLogins
+ * @returns {Object}
+ */
+function getMemberInviteOptions(
+    reports,
```
Comment (parasharrajat) [inline src/libs/OptionsListUtils.js:730]: Do we need this argument? We can just remove it and refactor all the usages of this function.

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/libs/OptionsListUtils.js (hunk 6 above)
Comment (metehanozyurt) [reply src/libs/OptionsListUtils.js]: Thanks @parasharrajat . I removed report parameter  and updated all  functions. Also removed `selectedOptions`.

**Review by parasharrajat — APPROVED** [review]
LGTM. Good work @metehanozyurt. Appreciate your patience.

cc: @roryabraham 

:ribbon: :eyes: :ribbon:  C+ reviewed

**Review by roryabraham — APPROVED** [review]

**Comment by roryabraham** [comment]
@Expensify/applauseleads can we be sure to update the regression test suite to account for the new sorting orders on these pages, and ensure that as part of the TC's we're verifying that these list pages are correctly sorted? Thanks 🙇
